### ISSUE TITLE
Quick Toggle

### ISSUE THREAD

**Issue by Mike-Reddick** [einaregilsson/Redirector#190]
Please add a setting to enable or disable the extension just with 1 left click on icon, and add a context menu entry to extension icon to "options". And add a keyboard shortcut to toggle extension also. Thanks!

**Comment by einaregilsson** [comment]
Thanks for the suggestion, but I prefer the way it is now :)

**Comment by Gitoffthelawn** [comment]
Just for reference, if anyone wants to add the context menu entry to the extension icon to edit Redirects, here is an example of another Web Extension that adds context menu items to it's toolbar button:

https://github.com/rNeomy/auto-tab-discard

And here are where the commands are listed in the manifest for that extension:
https://github.com/rNeomy/auto-tab-discard/blob/0b284d797fed3117cd391399777b20f0d22c5850/manifest.json#L78

**Comment by karlicoss** [comment]
Hey, wondering whether you are totally against the hotkey for the sake of simplicity? My usecase is being able to quickly toggle the extension on a one-off basis (to see the original page). Hotkey to toggle Redirector globablly + Ctrl-R to reload the page + hotkey to toggle it back would suit such usecase well.
At a quick glance, it seems that it would be just a matter of adding an `onCommand` callback and calling `updateIcon`, so hopefully won't increase the complexity and won't break anything. If you're okay with it, I'm happy to contribute and check against all browsers you support. Thanks!

**Comment by einaregilsson** [comment]
@karlicoss I'm not against it, but would probably want it to be an opt-in thing. E.g. you'd have to check "Enable on/off hotkey" in the settings and then the hotkey would be active. Reason being a hotkey can always conflict with another extension and I'm pretty sure as soon as we'd add one someone would send me a message saying it was conflicting with XYZ extension.  What do you a think a good hotkey would be?

**Comment by karlicoss** [comment]
Yeah, fair enough about conflicts. I already sort of struggle to pick non-conflicting hotkeys for extensions, so can't really come up with a good one.
I guess would be ideal if the default wasn't set at all, and those who want the hotkey could set it up themselves in the browser settings? However at least judging by the docs, it seems that manifest always requires some hotkey. I see a couple of options:
- I could try setting it to empty string anyway and seeing what happens, my hunch is that the manifest would still be valid
- if the above doesn't work/seems too fragile for your liking, we could set the hotkey to something ridiculous like `Ctrl-Alt-F12` (or something [similar](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/commands#Key_combinations)). Unlikely such a key would be set up for someone else, so it would basically be like a no-op.

**Comment by grahamperrin** [comment]
+1 to: 

* the option (_not_ a default) to disable or re-enable with a single-click on the toolbar icon
* a context menu for the toolbar icon. 
* _not_ having a shortcut by default
* Auto Tab Discard as exemplary, although re: https://github.com/rNeomy/auto-tab-discard/issues/188 Redirector does lend itself to the visual simplicity of **green**/enabled/_on_ and (without reducing contrast) **red**/disabled/_off_

**Comment by einaregilsson** [comment]
> I guess would be ideal if the default wasn't set at all, and those who want the hotkey could set it up themselves in the browser settings?

That would be nice, but also mean that very few would probably ever use the feature. But I'd guess if a hotkey conflicts with another extension both extensions get notified when it gets activated. We could just by default do nothing, and you'd have to check a "Use hotkey" thing to opt into the hotkey doing something.

**Comment by karlicoss** [comment]
> I'd guess if a hotkey conflicts with another extension both extensions get notified when it gets activated

If both are using [commands API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/commands/onCommand) (which is preferable to, say, `onpress` event callback), pretty sure the the first extension to define the hotkey wins. Sadly, none of the browsers make conflicts very explicit for the user unless they check in the settings themselves.

**Comment by einaregilsson** [comment]
Well, it looks like the user can change the suggested key if it is conflicting.

![Screenshot 2020-12-09 at 08 31 34](https://user-images.githubusercontent.com/557149/101604561-0052d300-39f9-11eb-88e2-8dff8cca7341.png)

So I guess it's mostly a matter of choosing a good default combo. 

https://developer.chrome.com/docs/extensions/reference/commands/ has a lot of good info about this.

Does anyone know if Firefox and Chrome are well aligned on this? Would it need any browser specific stuff?

**Comment by karlicoss** [comment]
> Well, it looks like the user can change the suggested key if it is conflicting.

Yep, these are the settings I meant :) 

> Would it need any browser specific stuff

It depends.. Chrome doesn't allow `Ctrl+Alt` as the doc says.. on the other hand in Firefox, many `Ctrl+Shift` bindings are reserved by the browser itself, and it's not possible to override (e.g. even if you don't use a certain default hotkey).

**Comment by Gitoffthelawn** [comment]
@einaregilsson Hi Einar. I read the paragraph you posted several times, and it sounds like that paragraph is perhaps referring to global shortcuts that are available outside of the Chrome browser (when the browser is not the active windows).

I'm not sure we would have any need for such a shortcut.  I think one just valid inside the browser itself is sufficient.

To my knowledge, all Firefox keyboard shortcuts are local to the browser, and not global to the OS.  Firefox has an excellent integrated UI to change all extension keyboard shortcuts.  Currently, however, you cannot redefine predefined shortcuts (I would love the ability to change Ctrl+D for creating a bookmark, as I sometimes hit that instead of Ctrl+F for the common Find operation!).

**Comment by mrx23dot** [comment]
It would be handy, I use it to redirect Youtube videos into embedded player, but not all videos allow that, so it takes 5clicks to temporary disable the rule.
